### PR TITLE
Replace CommCare translations autocomplete with atwho

### PR DESCRIPTION
**Pull request by orangejenny**
Borrowing heavily from https://github.com/dimagi/Vellum/pull/440

Odds are good that the atwho inclusions & options will go somewhere more general later, but for the moment this is the only place they're being used.

screen shot 2016-10-24 at 3 19 16 pm

@calellowitz / @emord

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 6 errors:

* corehq/apps/translations/static/translations/js/translations.js, line 63 - 'searchKey' is defined but never used (no-unused-vars)
* corehq/apps/translations/static/translations/js/translations.js, line 68 - 'should_startWithSpace' is defined but never used (no-unused-vars)
* corehq/apps/translations/static/translations/js/translations.js, line 71 - '$li' is defined but never used (no-unused-vars)
* corehq/apps/translations/static/translations/js/translations.js, line 86 - Missing trailing comma. (comma-dangle)
* corehq/apps/translations/static/translations/js/translations.js, line 90 - 'otherEvent' is defined but never used (no-unused-vars)
* corehq/apps/translations/static/translations/js/translations.js, line 95 - Missing trailing comma. (comma-dangle)

File: corehq/apps/translations/static/translations/js/translations.js [hunk 1]
```diff
@@ -53,37 +53,48 @@ var mk_translation_ui = function (spec) {
...
+                    maxLen: Infinity,
+                    suffix: "",
+                    tabSelectsMatch: false,
+                    callbacks: {
+                        filter: function(query, data, searchKey) {
+                            return _.filter(data, function(item) {
+                                return item.name.indexOf(query) !== -1;
+                            });
+                        },
+                        matcher: function(flag, subtext, should_startWithSpace) {
```
Comment (dimagimon) [inline corehq/apps/translations/static/translations/js/translations.js]: 'should_startWithSpace' is defined but never used (no-unused-vars)

File: corehq/apps/translations/static/translations/js/translations.js [hunk 2]
```diff
@@ -53,37 +53,48 @@ var mk_translation_ui = function (spec) {
...
+                            return $input.val();
+                        },
+                        beforeInsert: function(value, $li) {
```
Comment (dimagimon) [inline corehq/apps/translations/static/translations/js/translations.js]: '$li' is defined but never used (no-unused-vars)

File: corehq/apps/translations/static/translations/js/translations.js [hunk 3]
```diff
@@ -53,37 +53,48 @@ var mk_translation_ui = function (spec) {
...
+                $input.one('focus', function () {
+                    $.ajax({
+                        url: suggestionURL,
+                        data: {
+                            lang: translation_ui.lang,
+                            key: that.key.val(),
+                        },
+                        success: function (data) {
+                            options.data = data;
+                            $input.atwho(options).on("inserted.atwho", function(event, $li, otherEvent) {
```
Comment (dimagimon) [inline corehq/apps/translations/static/translations/js/translations.js]: 'otherEvent' is defined but never used (no-unused-vars)

**Comment by orangejenny** [comment]
fixed the missing commas, leaving the unused vars alone as they're part of the atwho callback signatures

File: corehq/apps/translations/static/translations/js/translations.js [hunk 4]
```diff
@@ -53,37 +53,47 @@ var mk_translation_ui = function (spec) {
...
+                                $input.val($input.data("selected-value")).change();
```
Comment (emord) [inline corehq/apps/translations/static/translations/js/translations.js]: Is this line still needed as well? Looks like what atwho would do on its own?

**Pull request merged**